internal/stoppropaganda/websites.go

```diff
@@ -275,6 +288,27 @@ func doSingleRequest(ws *Website, req *fasthttp.Request, resp *fasthttp.Response
 
 	resp.ShouldDiscardBody = true
 
+	if *flagUseQueryParams == 1 {
+		// clean up query string and cookies from previous request
+		ws.req.URI().SetQueryString("")
```

Comment:
Check function arguments - there is `req` passed already, which is copy of `ws.req`. Also `ws.req`
 is not used at all in this function.

A bit off topic - accessing `ws.req` like this is not thread-safe and leads to race conditions (app will crash). This is the reason why there is function argument `req` which is copy of `ws.req` which was copied in a thread-safe manner before calling this function.

Reply:
Updated